Assets/AnkrSDK/Examples/Scripts/UI/DisconnectButtonController.cs

```diff
@@ -14,49 +16,34 @@ public class DisconnectButtonController : MonoBehaviour
 
 		private void OnEnable()
 		{
-			SubscribeOnTransportEvents().Forget();
+			SubscribeEvents().Forget();
 
 			_button.onClick.AddListener(OnButtonClick);
 		}
 
 		private void OnDisable()
 		{
-			UnsubscribeFromTransportEvents();
+			UnsubscribeEvents().Forget();
 
 			_button.onClick.RemoveAllListeners();
 		}
 
-		private async UniTaskVoid SubscribeOnTransportEvents()
+		private async UniTaskVoid SubscribeEvents()
 		{
-			await UniTask.WaitUntil(() => WalletConnect.Session != null);
-
-			var walletConnectSession = WalletConnect.Session;
-			if (walletConnectSession == null)
-			{
-				return;
-			}
-
-			walletConnectSession.OnTransportConnect += UpdateDisconnectButtonState;
-			walletConnectSession.OnTransportDisconnect += UpdateDisconnectButtonState;
-			walletConnectSession.OnTransportOpen += UpdateDisconnectButtonState;
+			await UniTask.WaitUntil(() => WalletConnect.Status != WalletConnectStatus.Uninitialized);
+			WalletConnect.SessionStatusUpdated += OnSessionStatusUpdated;
 		}
 
-		private void UnsubscribeFromTransportEvents()
+		private async UniTaskVoid UnsubscribeEvents()
 		{
-			var walletConnectSession = WalletConnect.Session;
-			if (walletConnectSession == null)
-			{
-				return;
-			}
-
-			walletConnectSession.OnTransportConnect -= UpdateDisconnectButtonState;
-			walletConnectSession.OnTransportDisconnect -= UpdateDisconnectButtonState;
-			walletConnectSession.OnTransportOpen -= UpdateDisconnectButtonState;
+			await UniTask.WaitUntil(() => WalletConnect.Status != WalletConnectStatus.Uninitialized);
```

Comment:
Doesn't make sense to wait for wallet connect to become initialized on unsubscribe state - if it is uninitialized, we did not subscribe yet